timesketch/api/v1/resources/sigma.py

```diff
@@ -92,10 +120,253 @@ def get(self, rule_uuid):
+            abort(HTTP_STATUS_CODE_NOT_FOUND, f"ValueError {e}")
+
+        if not rule:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, "No rule found with this ID.")
+        return_rules = []
+
+        assert isinstance(rule, SigmaRule)
+
+        return_rules.append(
+            _enrich_sigma_rule_object(rule=rule))
+
+        meta = {
+            "current_user": current_user.username,
+        }
+        return jsonify({"objects": return_rules, "meta": meta})
+
+    @login_required
+    def delete(self, rule_uuid):
+        """Handles DELETE request to the resource.
+        Args:
+            rule_uuid: uuid of the rule
+        """
+
+        rule = SigmaRule.query.filter_by(rule_uuid=rule_uuid).first()
+
+        db_session.delete(rule)
+        db_session.commit()
+
+        return HTTP_STATUS_CODE_OK
+
+    @login_required
+    def put(self, rule_uuid):
+        """Handles update request to Sigma rules
+        Args:
+            rule_uuid: uuid of the rule
+        Returns:
+            The updated sigma object in JSON (instance of
+            flask.wrappers.Response)
+        """
+
+        try:
+            rule = SigmaRule.query.filter_by(rule_uuid=rule_uuid).first()
+
+        except Exception as e: # pylint: disable=broad-except
+            logger.error(
+                "Unable to get the Sigma rule",
+                exc_info=True,
+            )
+            abort(HTTP_STATUS_CODE_NOT_FOUND, f"ValueError {e}")
+
+        form = request.json
+        if not form:
+            form = request.data
+        if not form.validate_on_submit():
+            abort(HTTP_STATUS_CODE_BAD_REQUEST, "Unable to validate form data.")
+
+        abort(HTTP_STATUS_CODE_NOT_FOUND, "Method not implemented yet")
+
+        # TODO(jaegeral): complete this method
+        rule_yaml = form.get("rule_yaml", "")
+        parsed_rule = ts_sigma_lib.get_sigma_rule_by_text(rule_yaml)
+
+        logger.debug(rule_yaml + parsed_rule + rule)
+
+
+
+    @login_required
+    def post(self, rule_uuid=None):
+        """Handles POST request to the resource.
+        Args:
+            rule_uuid: uuid of the rule
+
+        Returns:
+            Sigma rule object and HTTP status code indicating
+            whether operation was sucessful.
+        """
+        form = request.json
+        if not form:
+            form = request.data
+
+        rule_yaml = form.get("rule_yaml", "")
+
+        if not rule_yaml:
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND,
+                "No rule_yaml supplied.",
+            )
+
+        parsed_rule = ts_sigma_lib.get_sigma_rule_by_text(rule_yaml)
+
+        if not rule_uuid:
+            rule_uuid = parsed_rule.get("id")
+
+        if not isinstance(rule_yaml, str):
+            abort(
+                HTTP_STATUS_CODE_FORBIDDEN, "rule_yaml needs to be a string."
+            )
+
+        # Query rules to see if it already exist
+        rule = SigmaRule.query.filter_by(rule_uuid=rule_uuid).first()
```

Comment:
Since you only use the `rule_uuid` to check if a rule with that ID already exist, you could save some data by just querying what you need from the db: `rule = SigmaRule.query(SigmaRule.rule_uuid).filter_by(rule_uuid=rule_uuid).first()`

Or you can make the whole check a bit easier:
```
if SigmaRule.query(SigmaRule.rule_uuid).filter_by(rule_uuid=parsed_rule.get("rule_uuid")).first() is None:
            abort(HTTP_STATUS_CODE_CONFLICT, "Rule already exist")
```

Reply:
I like that

Reply:
your query would fail with an ACL Error ```(Pdb) SigmaRule.query(SigmaRule.rule_uuid).filter_by(rule_uuid=rule_uuid).first()
*** TypeError: 'AclBaseQuery' object is not callable```

If the rule does not exist, so all the remaining code would be in a try except block, so I actually would like to exit only if the rule was found so I reverted it back to what it was before, lmk what you think

Reply:
The effort to change/debug this is probably not worth the potential benefits of only query the uuid instead of the full entry. Will mark it resolved.